contrib/neon/pagestore_smgr.c

```diff
@@ -1432,7 +1438,7 @@ zenith_truncate(SMgrRelation reln, ForkNumber forknum, BlockNumber nblocks)
 	 */
 	XLogFlush(lsn);
 
-	SetLastWrittenPageLSN(lsn);
+	SetLastWrittenLSN(lsn, reln->smgr_rnode.node.relNode, REL_METADATA_PSEUDO_BLOCKNO, REL_METADATA_PSEUDO_BLOCKNO);
```

Comment:
Hmm, so you don't remove the cache entries for the truncated blocks? Is that ok?

Reply:
I think that it is ok: LRU replacement algorithm is used to avoid cache overflow.
And if OID is reused for some other relation, then - see above: nothing critical can happen because of max(relid) assignment.